### PR TITLE
feat(terraform): - TF and CFN - Add a policy for ensuring AWS Bedrock Agent is encrypted with a CMK

### PR DESCRIPTION
**Pull request by inbalavital**
## Description

Add TF and CFN policy to check for the use of customer_encryption_key_arn in aws_bedrockagent_agent

## New policies
TF and CFN policy to check for the use of customer_encryption_key_arn in aws_bedrockagent_agent

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: tests/cloudformation/checks/resource/aws/test_BedrockAgentEncrypted.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import os
+import unittest
+
+from checkov.cloudformation.checks.resource.aws.BedrockAgentEncrypted import check
+from checkov.cloudformation.runner import Runner
+from checkov.runner_filter import RunnerFilter
+
+
+class TestBedrockAgentEncrypted(unittest.TestCase):
+
```
Comment (ChanochShayner) [inline tests/cloudformation/checks/resource/aws/test_BedrockAgentEncrypted.py:10]: Maybe removing this blank line (between the class and the func) will solve the weird cfn-lint issue

Comment (inbalavital) [reply tests/cloudformation/checks/resource/aws/test_BedrockAgentEncrypted.py:10]: It didn't help, the issue is that `cfn-lint` should run only over yml and json files, it fails on every python file I checked locally

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -43,16 +43,25 @@ jobs:
...
+          fi
+      - name: Install cfn-lint
+        if: env.RELEVANT_FILES_CHANGED == 'true'
         run: |
           pip install -U cfn-lint
       - name: Lint Cloudformation templates
-        if: steps.changed-files-specific.outputs.any_changed == 'true'
-        env:
-          ALL_CHANGED_FILES: ${{ steps.changed-files-specific.outputs.all_changed_files }}
+        if: env.RELEVANT_FILES_CHANGED == 'true'
```
Comment (maxamel) [inline .github/workflows/pr-test.yml:62]: Do we need to check this again?
Can this be run in the previous step where we already checked?

File: .github/workflows/pr-test.yml [hunk 3]
```diff
@@ -43,16 +43,25 @@ jobs:
...
           files: tests/cloudformation/checks/resource/aws/**/*
-      - name: Install cfn-lint
+      - name: Filter YAML and JSON files
         if: steps.changed-files-specific.outputs.any_changed == 'true'
+        id: filter-files
+        run: |
+          YAML_JSON_FILES=$(echo ${{ steps.changed-files-specific.outputs.all_changed_files }} | tr ' ' '\n' | grep -E '\.ya?ml$|\.json$' | tr '\n' ' ')
+          if [ -n "$YAML_JSON_FILES" ]; then
+            echo "YAML_JSON_FILES=$YAML_JSON_FILES" >> "$GITHUB_ENV"
+            echo "RELEVANT_FILES_CHANGED=true" >> "$GITHUB_ENV"
```
Comment (maxamel) [inline .github/workflows/pr-test.yml:53]: Isn't this redundant? RELEVANT_FILES_CHANGED
Why we're just not using YAML_JSON_FILES and iterating over it ? I think one env var should be enough here

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (inbalavital) [reply .github/workflows/pr-test.yml:62]: you're right, the 2 steps can be merged

File: .github/workflows/pr-test.yml (hunk 3 above)
Comment (inbalavital) [reply .github/workflows/pr-test.yml:53]: fixing

**Review by maxamel — APPROVED** [review]

**Pull request merged**